### PR TITLE
CLN: tests/window/moments

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them

Reduces a lot of no-op parameterizations by creating more specific fixtures

### CODE REVIEW

File: pandas/tests/window/moments/test_moments_consistency_expanding.py [hunk 1]
```diff
@@ -5,141 +5,131 @@
...
-        pass
-    else:
-        expanding_f_result = x.expanding(min_periods=min_periods).sum()
-        expanding_apply_f_result = x.expanding(min_periods=min_periods).apply(
-            func=f, raw=True
-        )
-        tm.assert_equal(expanding_f_result, expanding_apply_f_result)
+def test_expanding_apply_consistency_sum_nans(all_data, min_periods, f):
+    if f is np.sum and not no_nans(all_data):
+        pytest.xfail("np.sum has different behavior with NaNs")
```
Comment (jbrockmendel) [inline pandas/tests/window/moments/test_moments_consistency_expanding.py:15]: request.node.add_marker?

**Comment by jbrockmendel** [comment]
One comment, otherwise LGTM

**Comment by jbrockmendel** [comment]
BTW just curious, is there a gameplan for the xfails in test_online and test_expanding?

**Comment by mroeschke** [comment]
> BTW just curious, is there a gameplan for the xfails in test_online and test_expanding?

Hmm those xfails are just not implemented features with associated issues I think. I don't have a timeline for tackling those yet.

**Pull request merged**